src/realm/object-store/class.hpp

```diff
@@ -0,0 +1,56 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2023 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+#ifndef REALM_OS_CLASS_HPP
+#define REALM_OS_CLASS_HPP
+
+#include <realm/obj.hpp>
+#include "realm/parser/keypath_mapping.hpp"
+#include "realm/parser/query_parser.hpp"
+#include <memory>
+
+namespace realm {
+
+class ObjectSchema;
+class Realm;
+class Query;
+
+class Class {
+public:
+    Class(const std::shared_ptr<Realm>& r, StringData object_type);
```

Comment:
To me, it feels more natural to have a `Realm::get_class(name) -> Class` method. Alternatively, it may be even better to offer something like:

```cpp
Realm::get_classes() const -> std::vector<Class>;
Class::get_schema() const -> const ObjectSchema&;
```

Then we could completely replace the logic when opening a Realm that iterates the `Schema` (aka `std::vector<ObjectSchema>`) and gets a `TableRef` for each element by its `TableKey`. This would also avoid the O(logN) lookup of the `ObjectSchema` from the `Realm` in the constructor by just iterating the `Schema`.

Reply:
:+1: